### PR TITLE
React to HttpRequestError API changes

### LINKED ISSUE #89080: Revisit HttpRequestException.HttpRequestError type
We just merged a new API adding an `HttpRequestError` property to `HttpRequestException`.
```C#
public HttpRequestException(string? message, Exception? inner = null, HttpStatusCode? statusCode = null, HttpRequestError? httpRequestError = null);
public HttpRequestError? HttpRequestError { get; }
```
There is controversy over HttpRequestError, in particular that it's a `Nullable<HttpRequestError>` when HttpRequestError.Unknown also exists.  There are then two different ways to communicate that a meaningful HttpRequestError value could not be provided.

I propose the above be changed to instead be:
```C#
public HttpRequestException(HttpRequestError httpRequestError, string? message = null, Exception? inner = null, HttpStatusCode? statusCode = null);
public HttpRequestError HttpRequestError { get; }
```
- HttpRequestError parameter to the ctor is moved to be first
- message is given a nullable default value
- HttpRequestError property is made non-nullable

### PR DESCRIPTION
**Pull request by MihaZupan**
Fixes #89080

2bee08ff525c6dadfa839e19a1bc26c8a6195402 and eee6bba9327564028d85a961a765800dbd073dc4 are slight tweaks, the rest is mainly just swapping parameters around.

### CODE REVIEW

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs [hunk 1]
```diff
@@ -208,7 +208,7 @@ public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, lon
 
                 if (quicStream == null)
                 {
-                    throw new HttpRequestException(SR.net_http_request_aborted, null, RequestRetryType.RetryOnConnectionFailure);
+                    throw new HttpRequestException(HttpRequestError.Unknown, SR.net_http_request_aborted, null, RequestRetryType.RetryOnConnectionFailure);
```
Comment (stephentoub) [inline src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs:211]: Not specific to this PR, but I'm surprised some of these are "Unknown".  Are we missing some kind of "UserRequested" / "Canceled" / "Aborted" / etc. enum status?

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs [hunk 2]
```diff
@@ -252,18 +252,18 @@ public async Task<HttpResponseMessage> SendAsync(CancellationToken cancellationT
                 {
                     case Http3ErrorCode.VersionFallback:
                         // The server is requesting us fall back to an older HTTP version.
-                        throw new HttpRequestException(SR.net_http_retry_on_older_version, ex, RequestRetryType.RetryOnLowerHttpVersion);
+                        throw new HttpRequestException(HttpRequestError.Unknown, SR.net_http_retry_on_older_version, ex, RequestRetryType.RetryOnLowerHttpVersion);
```
Comment (stephentoub) [inline src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs:255]: After this lands, I think it'd be worthwhile auditing everything that's still Unknown to determine whether there are additional values we should be adding to the enum.

**Review by stephentoub — APPROVED** [review]
Thanks.

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs (hunk 1 above)
Comment (MihaZupan) [reply src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs:211]: I was considering whether some of these should be `ConnectionError`, but with the description of _"A transport-level failure occurred while connecting to the remote endpoint."_, it doesn't feel like the right fit.

A bunch of these are on retriable exceptions so in most cases the user won't see them, but it does feel like we're missing _something_.

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs (hunk 2 above)
Comment (antonfirsov) [reply src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs:255]: This is the reason why I opened #89097.

**Review by antonfirsov — APPROVED** [review]
Thanks a lot!

**Pull request merged**